### PR TITLE
android x support + upgrades

### LINKED ISSUE #12: Android X support
Any plans on creating a new branch for android x or for moving on to android x?

Generated code always produces app compat code (e.g. wrong `@NonNull` annotations)

Sadly, the jetifier does not work with generated code and therefore it does not transform it to android x...

If I see it correct, this is the only line in code that would need adjustment: https://github.com/beworker/featured/blob/master/featured-compiler/src/main/java/de/halfbit/featured/compiler/Names.java#L64

### PR DESCRIPTION
**Pull request by MFlisar**
I removed the app compat dependencies and replaced them with jetbrain dependencies

Additionally I did following:
* updated sdk version from 24 to 28 and also all app compat dependencies
* updated build tools, gradle, gradle wrapper
* extracted version definitions into own file
* minSdk changed to 14, because support version 28 forces this min sdk now
* removed app compat dependency from featured module (replaced `ArrayMap` with `HashMap`)

Fixes #12

### CODE REVIEW

File: featured-compiler/build.gradle [hunk 1]
```diff
@@ -13,18 +13,21 @@ for (File file : sdkHandler.sdkLoader.repositories) {
 }
 
 dependencies {
-    compile project(':featured-annotations')
-    compile deps.autoservice
-    compile deps.autocommon
-    compile deps.javapoet
+    compileOnly deps.android
```
Comment (sergejsha) [inline featured-compiler/build.gradle:16]: I had to replace this line with `testCompileOnly deps.android` to be able to run tests. 

The tests and `FeatureHost` stub class used by tests need some attention too. `android.support.annotation.NonNull` should be replaced with `org.jetbrains.annotations.NotNull`. Could you do it please?

**Review by sergejsha — CHANGES_REQUESTED** [review]
Very nice work! Thanks for submitting! Unit tests should run again (basically the same replace task should be applied on them) and we can merge it.

**Comment by sergejsha** [comment]
Would you also resolve the conflicts please? Not sure why there are some. I haven't merged any changes to master recently.

**Comment by MFlisar** [comment]
I forked your project recently (the day I created the last issue), not sure why my fork was some commits behind. Merged everything now and fixed all test cases.

**Pull request merged**